projects/noloan2/app/src/main/java/hasadna/noloan2/LawsuitPdfActivity.java

```diff
@@ -0,0 +1,327 @@
+
+        // Main LawsuitActivity path:
+        // app_name/lawsuit_main_dir/
+        lawsuitMainPath = Paths.get(Environment.getExternalStorageDirectory().getPath(),getString(R.string.app_name),LAWSUIT_MAIN_DIR_NAME).toString();
+        if (!dirExists(lawsuitMainPath)){
+            File directory = new File(lawsuitMainPath);
+            directory.mkdirs();
+        }
+
+        // Template's dir:
+        // app_name/lawsuit_main_dir/lawsuit_template_dir/
+        // TODO: Currently template's file is been stored in user's device storage. Check to see if this should be part of the apk package.
+        lawsuitTemplateDirPath = Paths.get(Environment.getExternalStorageDirectory().getPath(),getString(R.string.app_name),LAWSUIT_MAIN_DIR_NAME,LAWSUIT_TEMPLATE_DIR_NAME).toString();
+        if (!dirExists(lawsuitTemplateDirPath)){
+            File directory = new File(lawsuitTemplateDirPath);
+            directory.mkdirs();
+        }
+
+        // Template file's path:
+        try {
+            lawsuitTemplateFilePath = Paths.get(lawsuitTemplateDirPath, LAWSUIT_TEMPLATE_FILE_NAME).toString();
+        }
+        catch (Exception e){
+            Toast.makeText(this, "No template file found for the lawsuit",
+                    Toast.LENGTH_LONG).show();
+        }
+        // User's Lawsuit dir:
+        // app_name/lawsuit_output_dir/
+        lawsuitOutputPath = Paths.get(Environment.getExternalStorageDirectory().getPath(),getString(R.string.app_name)+" - "+ LAWSUIT_OUTPUT_DIR_NAME).toString();
+        if (!dirExists(lawsuitOutputPath)){
+            File directory = new File(lawsuitOutputPath);
+            directory.mkdirs();
+        }
+    }
+
+    /**
+     * Cehcks if a directory given in the parameter exists.
+     * @param path Directory to look for.
+     * @return True - Exists, False - Doesn't exists.
+     */
+    public boolean dirExists(String path)
+    {
+        boolean res = false;
+        File dir = new File(path);
+        if(dir.exists() && dir.isDirectory())
+            res = true;
+        return res;
+    }
+
+    // Check permissions: TODO: Check if this should be done at an earlier activity.
+    public boolean isReadStoragePermissionGranted() {
+        if (Build.VERSION.SDK_INT >= 23) {
+            if (checkSelfPermission(Manifest.permission.READ_EXTERNAL_STORAGE)
+                    == PackageManager.PERMISSION_GRANTED) {
+                Log.w(TAG, "Permission is granted1");
+                return true;
+            } else {
+
+                Log.v(TAG, "Permission is revoked1");
+                ActivityCompat.requestPermissions(this, new String[]{Manifest.permission.READ_EXTERNAL_STORAGE}, 3);
+                return false;
+            }
+        } else { //permission is automatically granted on sdk<23 upon installation
+            Log.v(TAG, "Permission is granted1");
+            return true;
+        }
+    }
+
+    public boolean isWriteStoragePermissionGranted() {
```

Comment:
The code for `isWriteStoragePermissionGranted()` is very similar to `isReadStoragePermissionGranted()`. Perhaps you could write a method that takes a parameter, and use only 1 method instead of 2?

Reply:
I removed the isReadStoragePermissionGranted() - as if write permission is granted -> read permission as well. And use of this activity requires the write permission.

https://developer.android.com/training/data-storage/files

Reply:
Great.